### ISSUE TITLE
Parsing proto Any fields as part of a bigger proto DataFrame

### ISSUE THREAD

**Issue by cinspire93** [scalapb/sparksql-scalapb#67]
Hi there @thesamet! Thanks again for helping with the previous issue. This is a continuation of the issue we discussed towards the end of the last issue. Assume we have the following protos:
```
message Event {
  bytes eventId = 1;
  google.protobuf.Any action = 2;
}

message Hit {
  bytes id = 1;
  string target = 2;
}
```
and I already have `eventDF` parsed from proto as follows:
```
val eventDF = srcDF.map { row => Event.parseFrom(row.getAs[Array[Byte]]("value") }.toDF

eventDF.printSchema
**prints**
root
 |-- event_id: binary (nullable = true)
 |-- action: struct (nullable = true)
 |    |-- type_url: string (nullable = true)
 |    |-- value: binary (nullable = true)
```
If I know from the `type_url` that it's going to be a `Hit`, how would I parse `action.value` and add it back to `eventDF`? I have tried your suggestion of using a UDF. However, I would get this error back:
```
java.lang.UnsupportedOperationException: Schema for type com.google.protobuf.ByteString is not supported
```
My current workaround to create a separate DataFrame just for `Hit`s, overwriting all `Hit.id` with `Event.eventId`, then joining the two DataFrames later on. This is very inefficient. Are there any other functionalities I am missing from scalapb?

**Comment by thesamet** [comment]
Thanks for reporting this. I wanted to update that I was able to reproduce this and it's a valid issue that needs to be solved. I will look into this.

**Comment by thesamet** [comment]
I added `ProtoSQL.udf` that can be used to create UDFs that return protos. There's an example on how to use it:

https://github.com/scalapb/sparksql-scalapb/blob/fa8c8532855b67f41ecc167ca14a86cbdce3406c/sparksql-scalapb/src/test/scala/PersonSpec.scala#L280-L295

You can use the same approach to parse `eventDF` from `srcDF` which could clean that code too.

**Comment by cinspire93** [comment]
Awesome sauce! Thanks for jumping on this so quickly :D Just to confirm this is only compatible with Scala 2.12 correct?

**Comment by thesamet** [comment]
Yes, ScalaPB 0.10.x only works with Scala 2.12 and up. The reason I removed Scala 2.11 is that the Java code that protobuf 3.11.x generates is incompatible with latest scalac 2.11 which reached EOL, so it became unmaintable.

I am still supporting 0.9.x while projects are being upgraded. I have backported this new feature into the 0.9.x and released it with sparksql-scalapb 0.9.1.

**Comment by cinspire93** [comment]
Much appreciated. EMR only supports up to 2.11 as of this moment. Their 6.0.x build, which has Scala 2.12, is still in beta. We'd be happy to migrate over once it's released. Thanks for making it compatible with 0.9.x.
